CartoDB/cartodb
# legends: custom: add support icons beyond svg

re: https://github.com/CartoDB/cartodb/issues/11371

currently only svg icons in custom legends will display properly, thus we had to update cartodb.js for this purpose

also, custom legends with icons from old editor won't appear properly, as they were saved with `url()` in the property value

a rake task to replace those icons in the custom legends for currently migrated legends as well as the migrator from editor to builder has been done

**acceptance test**

- create a custom legend with icon, both svg and png in editor, promote the user to builder and check the legends look OK
- set an icon with `url()` for a builder user, run the rake task and check the icon look properly now
- Test the rake task:
  - Create a builder map and add a legend. Manually send a request to set the legend icon to be wrapped inside `url(...)` instead of a plain URL. The image should be broken.
   - Publish the map
   - Run the rake `carto:legends:migration:migrate_custom_image`
   - Now, refreshing the builder and the embed, you should be able to see the images without needing to republish the map. Be careful with caches.
- In all cases, see that the icon renders correctly (see https://github.com/CartoDB/cartodb/pull/11391#issuecomment-276625838)

**Deploy**
After the deploy finishes, run the `carto:legends:migration:migrate_custom_image` rake in all clouds.
